### PR TITLE
NamespaceGlobalMethodSecurity groovy->java

### PR DESCRIPTION
**Pull request by jzheaux**
Note that the `WhenUsingAspectJ` tests are still simply verifying structure instead of behavior. This is because the project appearsto be misconfigured in some way such that AspectJ advice isn't getting woven in at runtime. The original Groovy tests also only verified structure and they may be that way for a similar reason.

Either way, I will open up a ticket so we can review why that is the case and if there is a good fix.

Issue: gh-4939

### CODE REVIEW

File: config/src/test/java/org/springframework/security/config/annotation/method/configuration/MethodSecurityServiceConfig.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+package org.springframework.security.config.annotation.method.configuration;
+
+import org.springframework.context.annotation.Bean;
+
+/**
+ * @author Josh Cummings
+ */
+public interface MethodSecurityServiceConfig {
```
Comment (rwinch) [inline config/src/test/java/org/springframework/security/config/annotation/method/configuration/MethodSecurityServiceConfig.java]: If you are going to extract this out, I'd do so as a class for consistency with the other tests. Then in the tests just register multiple configurations.

File: config/src/test/java/org/springframework/security/config/annotation/method/configuration/NamespaceGlobalMethodSecurityExpressionHandlerTests.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+			.doesNotThrowAnyException();
+
+		assertThatThrownBy(() -> this.service.postHasPermission("denied"))
+			.isInstanceOf(AccessDeniedException.class);
+	}
+
+	@EnableGlobalMethodSecurity(prePostEnabled = true)
+	public static class CustomAccessDecisionManagerConfig extends GlobalMethodSecurityConfiguration {
+		@Bean
+		public MethodSecurityServiceImpl service() {
```
Comment (rwinch) [inline config/src/test/java/org/springframework/security/config/annotation/method/configuration/NamespaceGlobalMethodSecurityExpressionHandlerTests.java]: Could this be removed an instead register MethodSecurityServiceConfig?

File: config/src/test/java/org/springframework/security/config/annotation/method/configuration/MethodSecurityServiceConfig.java (hunk 1 above)
Comment (jzheaux) [reply config/src/test/java/org/springframework/security/config/annotation/method/configuration/MethodSecurityServiceConfig.java]: great, will do

File: config/src/test/java/org/springframework/security/config/annotation/method/configuration/NamespaceGlobalMethodSecurityExpressionHandlerTests.java (hunk 2 above)
Comment (jzheaux) [reply config/src/test/java/org/springframework/security/config/annotation/method/configuration/NamespaceGlobalMethodSecurityExpressionHandlerTests.java]: Yes, just missed it.

**Comment by rwinch** [comment]
Thanks for the PR! This is now merged into master

**Pull request merged**